internal/monitors/conviva/accountsservice.go

```diff
@@ -0,0 +1,233 @@
+package conviva
+
+import (
+	"context"
+	"fmt"
+	"golang.org/x/sync/errgroup"
+	"sync"
+	"time"
+)
+
+const (
+	accountsURLFormat             = "https://api.conviva.com/insights/2.4/accounts.json"
+	filtersURLFormat              = "https://api.conviva.com/insights/2.4/filters.json?account=%s"
+	metricLensDimensionsURLFormat = "https://api.conviva.com/insights/2.4/metriclens_dimension_list.json?account=%s"
+)
+
+// account for Conviva account data
+type account struct {
+	id                   string
+	Name                 string
+	filters              map[string]string
+	metricLensFilters    map[string]string
+	metricLensDimensions map[string]float64
+}
+
+// accountsService interface for accounts related methods
+type accountsService interface {
+	getDefault() (*account, error)
+	getMetricLensDimensionMap(accountName string) (map[string]float64, error)
+	getID(accountName string) (string, error)
+	getFilters(accountName string) (map[string]string, error)
+	getMetricLensFilters(accountName string) (map[string]string, error)
+	getFilterID(accountName string, filterName string) (string, error)
+	getMetricLensDimensionID(accountName string, metricLensDimension string) (float64, error)
+}
+
+type accountsServiceImpl struct {
+	defaultAccount      *account
+	accounts            map[string]*account
+	ctx                 context.Context
+	timeout             *time.Duration
+	client              httpClient
+	mutex               *sync.RWMutex
```

Comment:
If you make mutex a value you don't need to initialize it in `newAccountsService`.  The default value for mutex is an unlocked mutex.